src/Workleap.AspNetCore.Authentication.ClientCredentialsGrant/OpenAPI/SecurityRequirementOperationFilter.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Globalization;
+using System.Text;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Workleap.AspNetCore.Authentication.ClientCredentialsGrant.OpenAPI;
+
+internal sealed class SecurityRequirementOperationFilter : IOperationFilter
+{
+    private readonly string? _audience;
+
+    public SecurityRequirementOperationFilter(IOptionsMonitor<JwtBearerOptions> jwtOptionsMonitor)
+    {
+        var jwtOptions = jwtOptionsMonitor.Get(ClientCredentialsDefaults.AuthenticationScheme);
+        this._audience = jwtOptions.Audience;
+    }
+    
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        var attributes = SwaggerUtils.GetRequiredPermissions(context.ApiDescription).ToArray();
+        if (attributes.Length == 0)
+        {
+            return;
+        }
+
+        // Method need to be idempotent since minimal api are preserving the state
+        AddAuthzErrorResponse(operation);
+        this.AddOperationSecurityReference(operation, attributes);
+        AppendScopeToOperationSummary(operation, attributes);
+    }
+    
+    private static void AddAuthzErrorResponse(OpenApiOperation operation)
+    {
+        operation.Responses.TryAdd(StatusCodes.Status401Unauthorized.ToString(CultureInfo.InvariantCulture), new OpenApiResponse { Description = ReasonPhrases.GetReasonPhrase(StatusCodes.Status401Unauthorized) });
```

Comment:
when it is generated, wonder if it has a ProblemDetails object? Also, do we want to support returning 401 with a string?

Reply:
Seems it is just returning an empty body. Also it is how it was implemented in the Workleap Platform.